### ISSUE TITLE
[no-type-alias] Conditional types are unhandled

### ISSUE THREAD

**Issue by whatisaphone** [typescript-eslint/typescript-eslint#949]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-type-alias": ["error", {"allowAliases": "always"}]
  }
}
```

```TS
type MyType<T> = T extends number ? number : null;
```

**Expected Result**

The rule should deem this line acceptable.

**Actual Result**

The rule always flags with "Type unhandled are not allowed"

**Additional Info**

This makes me sad.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`        | `2.1.0` |
| `TypeScript`                       | `3.5.1` |
| `ESLint`                           | `6.3.0` |
| `node`                             | `8.10.0` |
| `npm`                              | `5.6.0` |

**Comment by otofu-square** [comment]
I'll look into it :eyes:
